src/ekat/ekat_physical_constants.hpp

```diff
@@ -0,0 +1,105 @@
+#ifndef EKAT_PHYSICAL_CONSTANTS
+#define EKAT_PHYSICAL_CONSTANTS
+
+#include "ekat/util/ekat_units.hpp"
+
+namespace ekat {
+
+template <typename RealType=double>
+class PhysicalConstant {
+  public:
+    enum Op {Times, Divide};
+
+    // no default
+    PhysicalConstant() = delete;
+
+    PhysicalConstant(const PhysicalConstant<RealType>&) = default;
+
+    // Constructor for single value (i.e., NIST std)
+    PhysicalConstant(const RealType& nvalue, const ekat::units::Units& units, const std::string& name) :
+      m_nist_val(nvalue), m_e3sm_val(nvalue), m_units(units), m_name(name), m_info("") {}
+
+    // Constructor for single value (i.e., NIST std)
+    PhysicalConstant(const RealType& nvalue, const ekat::units::Units& units, const std::string& name,
+      const std::string& info) :
+      m_nist_val(nvalue), m_e3sm_val(nvalue), m_units(units), m_name(name), m_info(info) {}
+
+
+    // Constructor for two different values (i.e., when E3SM value and NIST value are different)
+    PhysicalConstant(const RealType& evalue, const RealType& nvalue, const ekat::units::Units& units,
+      const std::string& name) : m_e3sm_val(evalue), m_nist_val(nvalue), m_units(units),
+      m_name(name), m_info("") {}
+
+    // Constructor for two different values (i.e., when E3SM value and NIST value are different)
+    PhysicalConstant(const RealType& evalue, const RealType& nvalue, const ekat::units::Units& units,
+      const std::string& name, const std::string& info) : m_e3sm_val(evalue), m_nist_val(nvalue), m_units(units),
+      m_name(name), m_info(info) {}
+
+
+    // Constructor for a constant that is defined by the product or quotient of 2 other constants.
+    PhysicalConstant(const PhysicalConstant<RealType>& lhs, const Op& op, const PhysicalConstant<RealType>& rhs,
+       const std::string& name) :  m_e3sm_val(lhs.m_e3sm_val), m_nist_val(lhs.m_nist_val),
+       m_units(lhs.m_units), m_name(name), m_info(lhs.m_info + "; " + rhs.m_info) {
+        switch (op) {
+          case (Times) : {
+            m_e3sm_val *= rhs.m_e3sm_val;
+            m_nist_val *= rhs.m_nist_val;
+            m_units = lhs.m_units * rhs.m_units;
+            break;
+          }
+          case (Divide) : {
+            m_e3sm_val /= rhs.m_e3sm_val;
+            m_nist_val /= rhs.m_nist_val;
+            m_units = lhs.m_units / rhs.m_units;
+            break;
+          }
+        }
+    }
+
+    std::string name() const {return m_name;}
+
+    std::string get_string() const {return m_name + " " + to_string(m_units) +
+      (m_info.empty() ? "" : " (" + m_info + ")");}
+
+    std::string unit_string() const {return to_string(m_units);}
```

Comment:
Why not simply add `get_units()`, and let the user decide what to do with it? If they want the units stirng, they can always call `to_string(m_units)` (or other ways of displaying units). But returning units also allows to create other quantities with compatible units.

Reply:
I meant to do that 👍